### ISSUE TITLE
[Feather] Incorrect GM1048 error says 'other' does not resolve when used in a function in a nested struct

### ISSUE THREAD

**Issue by KormexGit** [YoYoGames/GameMaker-Bugs#2103]
### Description
If you use other inside a function that's in a struct inside another struct, feather will give an error saying other does not resolve to anything in this scope. Actually running the game shows otherwise, as the code works just fine. In fact, feather itself shows that other is an instance. 
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/100942509/2627207e-c06b-4ab3-8850-b3cfc79c29ed)

### Steps To Reproduce
```
a = 0;
struct = {
	subStruct : {
		func : function() {
			other.a = 5;	//GM1048 error here
			show_debug_message(other.a); //debug message appears as 5 when func is called
		}
	}
}
struct.subStruct.func();
show_debug_message(a); //also appears as 5
```

[4d06c8b7-01c3-401c-b107-7a6ef0825236](https://api.gamemaker.io/api/github/downloads/4d06c8b7-01c3-401c-b107-7a6ef0825236)

**Comment by zreedy** [comment]
The provided example does not demonstrate the issue. When the call is from the Create event this error should show as there's exactly 1 call site and `other` is indeed unbound at this call site.
However, the issue is present as documented if the function call is cut/pasted to a collision event. Change to check LocalContext is Event in addition to SelfContext. Not a perfect fix but will catch the common case where `other` is used outside of collision events or with statements.

A future improvement (needed for other features too) would be to check `other`'s binding at the call-site and highlight the function call instead.

**Comment by alicemoretti** [comment]
Verified fixed on Beta IDE v2023.1100.0.435 Beta Runtime v2023.1100.0.448.

Thank you
